gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -594,7 +594,7 @@ public URL read(JsonReader in) throws IOException {
             return null;
           }
           String nextString = in.nextString();
-          return "null".equals(nextString) ? null : new URL(nextString);
+          return nextString.equals("null") ? null : new URL(nextString);
```

Comment:
I also think `nextString` should never return `null` (if there was a JSON null, it would throw `IllegalStateException` due to token mismatch).

Using `Objects.equals` here makes a strong suggestion that the value can be `null`. But if that would be the case, the check `Objects.equals(null, "null")` would be `false` and it would call `new URL(null)` which on JDK 17 throws a `MalformedURLException`.

Would it be fine then to keep these changes?

Reply:
OK, the `new URL(null)` argument is convincing. :-)